src/compiler/llvm/mir.rs

```diff
@@ -0,0 +1,215 @@
+        table: &'ctx StringTable,
+    ) -> Self {
+        let module = ctx.create_module(module);
+
+        // Create a function to build
+        let ft = ctx.void_type().fn_type(&[], false);
+        let name = table.get(func_name).unwrap();
+        let function = module.add_function(&name, ft, None);
+
+        let builder = ctx.create_builder();
+
+        Self {
+            context: ctx,
+            module,
+            builder,
+            table,
+            function,
+            vars: HashMap::default(),
+        }
+    }
+
+    fn to_label(&self, vd: &VarDecl) -> String {
+        let name = self.table.get(vd.name()).unwrap();
+        let scope = vd.scope();
+        format!("{}_{}", name, scope)
+    }
+}
+
+impl<'ctx> Transformer<PointerValue<'ctx>, BasicValueEnum<'ctx>> for LlvmTransformer<'ctx> {
+    fn start_bb(&mut self, bb: BasicBlockId) {
+        let bb = self
+            .context
+            .append_basic_block(self.function, &bb.to_string());
+        self.builder.position_at_end(bb);
+    }
+
+    fn add_var(&mut self, id: VarId, decl: &VarDecl) {
+        let name = self.to_label(decl);
+
+        // Check if variable name already exists
+        match self.vars.entry(id) {
+            // If it does -> return an error
+            Entry::Occupied(_) => {
+                panic!("Attempting to add a variable that is already in the table")
+            }
+
+            // If not, then allocate a pointer in the Builder
+            Entry::Vacant(ve) => {
+                // and add a mapping from VarID to the pointer in the local var table
+                let ty = self.context.i64_type();
+                let ptr = self.builder.build_alloca(ty, &name);
+                ve.insert(ptr);
+            }
+        }
+    }
+
+    fn add_temp(&mut self) {
+        todo!()
+    }
+
+    fn term_return(&mut self) {
+        self.builder.build_return(None);
+    }
+
+    fn assign(&mut self, span: Span, l: PointerValue<'ctx>, v: BasicValueEnum<'ctx>) {
+        self.builder.build_store(l, v);
+    }
+
+    fn var(&self, v: VarId) -> PointerValue<'ctx> {
+        *self
+            .vars
+            .get(&v)
+            .expect("Cound not find given VarId in vars table")
+    }
+
+    fn constant(&self, c: Constant) -> BasicValueEnum<'ctx> {
+        match c {
+            Constant::Unit => todo!(),
+            Constant::I8(_) => todo!(),
+            Constant::I16(_) => todo!(),
+            Constant::I32(_) => todo!(),
+            Constant::I64(i) => self.context.i64_type().const_int(i as u64, true).into(),
+            Constant::U8(_) => todo!(),
+            Constant::U16(_) => todo!(),
+            Constant::U32(_) => todo!(),
+            Constant::U64(_) => todo!(),
+            Constant::F64(_) => todo!(),
+            Constant::Bool(_) => todo!(),
+            Constant::StringLiteral(_) => todo!(),
+            Constant::Null => todo!(),
+            Constant::SizeOf(_) => todo!(),
+        }
+    }
```

Comment:
To continue the MIR structure exists only in the MIR Traverser, I think that the transformer should have a function for each type rather than one function that takes a `Constant` enum.